lib/resolveTaskFn.js

```diff
@@ -99,10 +111,23 @@ module.exports = function resolveTaskFn({
   }
   debug('execaOptions:', execaOptions)
 
+  // Assign the final taskFn to be executed later
+  let taskFn
+
+  if (shell) {
+    // if the command was a function, it's already evaluated and should include file names.
+    // Otherwise, escape filenames for non-functional tasks to prevent issues
+    const shellString = isFn ? command : getShellString(command, files)
+    debug('Warning! Using the shell option with resolved command string:')
+    debug('%s', shellString)
+    taskFn = () => execa.command(shellString, execaOptions)
+  } else {
+    // execa and cross-spawn handle escaping of the arguments
+    taskFn = () => execa(cmd, isFn ? args : args.concat(files), execaOptions)
+  }
+
   return async (ctx = getInitialState()) => {
-    const result = await (shell
-      ? execa.command(isFn ? command : `${command} ${files.join(' ')}`, execaOptions)
-      : execa(cmd, isFn ? args : args.concat(files), execaOptions))
+    const result = await taskFn()
```

Comment:
Create the `taskFn` at the resolve phase outside this callback, instead of later when actually running the task.